code/processes/wdb.q

```diff
@@ -123,14 +127,34 @@ upserttopartition:{[dir;tablename;tabdata;pt;expttype;expt]
 	/-key in partsizes are directory to partition, need to drop training slash in directory key
 	.merge.partsizes[first ` vs directory]+:(count r;-22!r);
 	};
+
+/- function to upsert to specified directory using enumerated extra partitioning
+upserttopartitionenum:{[dir;tablename;tabdata;pt;expttype;expt]
+	hdbsym:` sv hdbsettings[`hdbdir],`sym;
+	/- enumerate current extra partition agains the hdb sym file
+	i:get[hdbsym]?hdbsym?first expt;
+	.lg.o[`save;"saving ",(string tablename)," data to partition ",
+				/- create directory location for selected partiton
+				string directory:` sv .Q.par[dir;pt;`$string i],tablename,`];
+	/- upsert selected data matched on partition to specific directory
+	.lg.o[`save;"directory: ",string directory];
+	.[
+	 upsert;
+	 (directory;r:?[tabdata;{(x;y;(),z)}[in;;]'[expttype;expt];0b;()]);
+	 {[e] .lg.e[`savetablesbypartenum;"Failed to save table to disk : ",e];'e}
+	 ];
```

Comment:
I think it's a little more readable if we define `r` outside of the apply (and add a comment to explain what's happening in the functional select)

```suggestion
	// select from tabdata where expttype in expt
	r:?[tabdata;{(x;y;(),z)}[in;;]'[expttype;expt];0b;()];
	.[upsert;(directory;r);
	 {[e] .lg.e[`savetablesbypartenum;"Failed to save table to disk : ",e];'e}];
```

I'm also wondering about the each-both - from line 135 I'm assuming we're only dealing with one unique partition value here, so do we need the iteration? could it just be something like this?

```q
r:?[tabdata;(=;expttype;first expt);0b;()];
```